opm/autodiff/StandardWellsDense_impl.hpp

```diff
@@ -2879,4 +3002,38 @@ namespace Opm {
         return potentials;
     }
 
+    template<typename TypeTag>
+    double
+    StandardWellsDense<TypeTag>::
+    wsolvent(const int well_index) const {
+
+        if (!has_solvent_) {
+            return 0.0;
+        }
+
+        // loop over all wells until we find the well with the matching name
+        for (const auto&  well : wells_ecl_) {
+            if (well->getStatus( current_timeIdx_ ) == WellCommon::SHUT) {
+                continue;
+            }
+
+            WellInjectionProperties injection = well->getInjectionProperties(current_timeIdx_);
+            if (injection.injectorType == WellInjector::GAS) {
+
+                double solventFraction = well->getSolventFraction(current_timeIdx_);
+
+                // Look until we find the correct well
+                if (well->name() == wells().name[well_index]) {
+                    return solventFraction;
+                }
+            }
+        }
+        // we didn't find it return 0;
+        // or should we throw?
```

Comment:
If we reach this point there is a mismatch between the well objects, I think we should do
```
assert(false); // Should never reach this point
```
before the return statement.